pyscriptjs/tests/integration/support.py

```diff
@@ -279,6 +384,69 @@ def wait_for_pyscript(self, *, timeout=None, check_js_errors=True):
         # events aren't being triggered in the tests.
         self.page.wait_for_timeout(100)
 
+    def _parse_py_config(self, doc):
+        configs = re.findall("<py-config>(.*?)</py-config>", doc, flags=re.DOTALL)
+        configs = [cfg.strip() for cfg in configs]
+        if len(configs) == 0:
+            return None
+        elif len(configs) == 1:
+            return toml.loads(configs[0])
+        else:
+            raise AssertionError("Too many <py-config>")
+
+    def _inject_execution_thread_config(self, snippet, execution_thread):
+        """
+        If snippet contains already a py-config, let's try to inject
+        execution_thread automatically. Note that this works only for plain
+        <py-config> with inline config: type="json" and src="..." are not
+        supported by this logic, which should remain simple.
+        """
+        cfg = self._parse_py_config(snippet)
+        if cfg is None:
+            # we don't have any <py-config>, let's add one
+            py_config_maybe = f"""
+            <py-config>
+                execution_thread = "{execution_thread}"
+            </py-config>
+            """
+        else:
+            cfg["execution_thread"] = execution_thread
+            dumped_cfg = toml.dumps(cfg)
+            new_py_config = f"""
+            <py-config>
+                {dumped_cfg}
+            </py-config>
+            """
+            snippet = re.sub(
+                "<py-config>.*</py-config>", new_py_config, snippet, flags=re.DOTALL
+            )
+            # no need for extra config, it's already in the snippet
+            py_config_maybe = ""
+        #
+        return snippet, py_config_maybe
+
+    def _pyscript_format(self, snippet, *, execution_thread, extra_head=""):
+        if execution_thread is None:
+            py_config_maybe = ""
```

Comment:
Is `execution_thread` ever going to be `None`?

Further, if it is indeed `None`, do we assume that `snippet` already has `<py-config>` present in it? I guess not?

Reply:
I am not sure if this `None` value for `execution_thread` is useful, but if I am missing something, maybe we should add a test for it?

Reply:
Alternatively, I guess this means that `injection` doesn't happen. Which means the parameter `execution_thread` may or may not be present in `<py-config>`. Then, if not present, during the merging process of configs for default values, I guess we default to using `main` as the value of `execution_thread`, is this understanding correct?

Reply:
> Is execution_thread ever going to be None?

yes, for example here:
https://github.com/pyscript/pyscript/blob/66145ea31ef2b10738645bb0216061e0ab354cfc/pyscriptjs/tests/integration/test_00_support.py#L9-L11

The point is that for the vast majority of tests, we want to run them twice: once in `main`, and once in `worker.
However, there are a few cases in which running them twice doesn't make sense. E.g.:
1. `test_00_support.py`, whose tests don't even mention pyscript 
2. `test_py_config.py`, where we test that we can parse the config, but we don't really care where the code is executed (because the config parsing happens earlier than the main/worker split)

So, with the current solution, by default we run test twice, but you can explicitly disable it.

> I guess we default to using main as the value of execution_thread, is this understanding correct?

yes, but the point is that we use `@with_execution_thread(None)` specifically on those tests in which the default value doesn't matter.